index.js

```diff
@@ -92,51 +92,72 @@ function getDelimiter(moduleId, symbol, parser) {
 }
 
 exports.defineTags = function (dictionary) {
-  ['type', 'typedef', 'property', 'return', 'param', 'template'].forEach(
-    function (tagName) {
-      const tag = dictionary.lookUp(tagName);
-      const oldOnTagText = tag.onTagText;
-      tag.onTagText = function (tagText) {
-        if (oldOnTagText) {
-          tagText = oldOnTagText.apply(this, arguments);
-        }
-        // Replace `templateliteral` with 'templateliteral'
-        const startIndex = tagText.search('{');
-        if (startIndex === -1) {
-          return tagText;
-        }
-        const len = tagText.length;
-        let open = 0;
-        let i = startIndex;
-        while (i < len) {
-          switch (tagText[i]) {
-            case '\\':
-              // Skip escaped character
-              ++i;
-              break;
-            case '{':
-              ++open;
-              break;
-            case '}':
-              if (!--open) {
-                return (
-                  tagText.slice(0, startIndex) +
-                  tagText
-                    .slice(startIndex, i + 1)
-                    .replace(/`([^`]*)`/g, "'$1'") +
-                  tagText.slice(i + 1)
-                );
-              }
-              break;
-            default:
-              break;
-          }
-          ++i;
+  const tags = [
+    'type',
+    'typedef',
+    'property',
+    'return',
+    'param',
+    'template',
+    'default',
+    'member',
+  ];
+
+  tags.forEach(function (tagName) {
+    const tag = dictionary.lookUp(tagName);
+    const oldOnTagText = tag.onTagText;
+    tag.onTagText = function (tagText) {
+      if (oldOnTagText) {
+        tagText = oldOnTagText.apply(this, arguments);
+      }
+
+      const startIndex = tagText.search('{');
+      if (startIndex === -1) {
+        return tagText;
+      }
+      const len = tagText.length;
+      let open = 0;
+      let i = startIndex;
+      while (i < len) {
+        switch (tagText[i]) {
+          case '\\':
+            // Skip escaped character
+            ++i;
+            break;
+          case '{':
+            ++open;
+            break;
+          case '}':
+            if (!--open) {
+              return (
+                tagText.slice(0, startIndex) +
+                tagText
+                  .slice(startIndex, i + 1)
+                  // Replace `templateliteral` with 'templateliteral'
+                  .replace(/`([^`]*)`/g, "'$1'")
+                  // Interface style semi-colon separators to commas
+                  .replace(/;/g, ',')
+                  // Remove trailing commas in object types
+                  .replace(/,(\s*\})/g, '$1')
```

Comment:
These two are too generic. Either remove them, or make them specific enough to match e.g. `{foo: string; bar: number; baz: boolean}`, but not `{"foo: string; bar: number; baz: boolean"}`

Reply:
I've changed these to a more in-depth solution that accounts for strings